lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/config/LndInstance.scala

```diff
@@ -11,22 +11,20 @@ import java.net._
 import java.nio.file._
 import scala.util.Properties
 
-case class LndInstance(
-    datadir: Path,
-    network: BitcoinNetwork,
-    listenBinding: URI,
-    restUri: URI,
-    rpcUri: URI,
-    bitcoindAuthCredentials: PasswordBased,
-    bitcoindRpcUri: URI,
-    zmqConfig: ZmqConfig,
-    debugLevel: LogLevel) {
-
-  lazy val certFile: File =
-    datadir.resolve("tls.cert").toFile
+sealed trait LndInstance {
+  def network: BitcoinNetwork
+  def listenBinding: URI
+  def restUri: URI
+  def rpcUri: URI
+  def bitcoindAuthCredentials: PasswordBased
+  def bitcoindRpcUri: URI
+  def zmqConfig: ZmqConfig
+  def debugLevel: LogLevel
 
   private var macaroonOpt: Option[String] = None
 
+  def datadir: Path
+
   def macaroon: String = {
```

Comment:
`macaroon` shouldn't be defined for the trait, and for `LndInstanceLocal` it should have the current definition. For remote it should need to be input

Reply:
same for `certFile`